pkg/services/store/entity_events.go

```diff
@@ -0,0 +1,117 @@
+package store
+
+import (
+	"context"
+	"time"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+type EntityEventType string
+
+const (
+	EntityEventTypeDelete EntityEventType = "delete"
+	EntityEventTypeCreate EntityEventType = "create"
+	EntityEventTypeUpdate EntityEventType = "update"
+)
+
+type EntityEvent struct {
+	Id        int64
+	EventType EntityEventType
+	Grn       string
+	Created   time.Time
+}
+
+type SaveEventCmd struct {
+	Grn       string
+	EventType EntityEventType
+}
+
+// EntityEventsService is a temporary solution to support change notifications in an HA setup
+// With this service each system can query for any events that have happened since a fixed time
+type EntityEventsService interface {
+	SaveEvent(ctx context.Context, cmd SaveEventCmd) error
+	GetLastEvent(ctx context.Context) (*EntityEvent, error)
+	GetAllEventsAfter(ctx context.Context, id int64) ([]*EntityEvent, error)
+
+	Run(ctx context.Context) error
+
+	deleteEventsOlderThan(ctx context.Context, duration time.Duration) error
+}
+
+func ProvideEntityEventsService(cfg *setting.Cfg, sqlStore *sqlstore.SQLStore) EntityEventsService {
+	return &entityEventService{
+		sql: sqlStore,
+		log: log.New("entity-events"),
+	}
+}
+
+type entityEventService struct {
+	sql *sqlstore.SQLStore
+	log log.Logger
+}
+
+func (e entityEventService) SaveEvent(ctx context.Context, cmd SaveEventCmd) error {
```

Comment:
Maybe add pointer to receiver? I think this can help avoiding problems in the future is entityEventService will have some non-reference type.